examples/stable_diffusion_v2/train_tuneavideo.py

```diff
@@ -0,0 +1,467 @@
+"""
+Tune A Video based on Stable Diffusion
+"""
+import argparse
+import importlib
+import logging
+import os
+import shutil
+
+import yaml
+from ldm.data.dataset_tuneavideo import load_data
+from ldm.modules.logger import set_logger
+from ldm.modules.train.callback import EvalSaveCallback, OverflowMonitor
+from ldm.modules.train.checkpoint import resume_train_network
+from ldm.modules.train.ema import EMA
+from ldm.modules.train.lr_schedule import create_scheduler
+from ldm.modules.train.optim import build_optimizer
+from ldm.modules.train.parallel_config import ParallelConfig
+from ldm.modules.train.tools import set_random_seed
+from ldm.modules.train.trainer import TrainOneStepWrapper
+from ldm.util import count_params, is_old_ms_version, str2bool
+from omegaconf import OmegaConf
+from utils.download import download_checkpoint
+
+import mindspore as ms
+from mindspore import Model, context, load_checkpoint, load_param_into_net
+from mindspore.communication.management import get_group_size, get_rank, init
+from mindspore.nn.wrap.loss_scale import DynamicLossScaleUpdateCell
+from mindspore.train.callback import LossMonitor, TimeMonitor
+
+os.environ["HCCL_CONNECT_TIMEOUT"] = "6000"
+
+logger = logging.getLogger(__name__)
+
+_version_cfg = {
+    "2.1": ("sd_v2-1_base-7c8d09ce.ckpt", "v2-inference.yaml", 512),
+    "2.1-v": ("sd_v2-1_768_v-061732d1.ckpt", "v2-vpred-inference.yaml", 768),
+    "2.0": ("sd_v2_base-57526ee4.ckpt", "v2-inference.yaml", 512),
+    "2.0-v": ("sd_v2_768_v-e12e3a9b.ckpt", "v2-vpred-inference.yaml", 768),
+    "1.5": ("sd_v1.5-d0ab7146.ckpt", "v1-inference.yaml", 512),
+    "1.5-wukong": ("wukong-huahua-ms.ckpt", "v1-inference-chinese.yaml", 512),
+}
+_URL_PREFIX = "https://download.mindspore.cn/toolkits/mindone/stable_diffusion"
+_MIN_CKPT_SIZE = 4.0 * 1e9
+
+
+def init_env(args):
```

Comment:
replace with `examples.stable_diffusion_v2.common.init_env` for easier refactoring later.

Reply:
I tried to use `examples.stable_diffusion_v2.common.init_env`, but it returns a different `rank_id` compared with the self-defined `init_env`. The former returns the current device id, while the latter always returns `0`.

Using `examples.stable_diffusion_v2.common.init_env`leads to an error in dataloader collection function, which I believe has something to do with the rank id.

Reply:
Fixed.